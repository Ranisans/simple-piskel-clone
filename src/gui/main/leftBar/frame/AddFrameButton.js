import React from 'react';
import { shortcuts } from '../../../../logic/shortcuts';

const AddFrameButton = ({ callback }) => (
  <div
    className='add_frame'
    onClick={callback}
    title={`shortcut: ${shortcuts.addFrame}`}
  >
    <div className='add_frame-icon'></div>
    <p className="add_frame-label">Add new frame</p>
  </div>
);

export default AddFrameButton;
