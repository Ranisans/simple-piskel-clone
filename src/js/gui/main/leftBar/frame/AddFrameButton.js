import React from 'react';

const AddFrameButton = ({ callback }) => (
  <div className='add_frame' onClick={callback}>
    <div className='add_frame-icon'></div>
    <p className="add_frame-label">Add new frame</p>
  </div>
);

export default AddFrameButton;
