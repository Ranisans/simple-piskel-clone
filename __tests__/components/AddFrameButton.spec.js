import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AddFrameButton from '../../src/gui/main/leftBar/frame/AddFrameButton';

describe('AddFrameButton tests', () => {
  it('create snapshot', () => {
    const component = shallow(
      <AddFrameButton callback={() => { }} />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it('calling callback', () => {
    const fakeCallback = jest.fn();
    const component = shallow(
      <AddFrameButton callback={fakeCallback} />,
    );
    component.find('div.add_frame').simulate('click');
    expect(fakeCallback).toHaveBeenCalledTimes(1);
  });
});
