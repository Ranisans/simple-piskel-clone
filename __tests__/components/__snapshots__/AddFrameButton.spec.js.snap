// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFrameButton tests create snapshot 1`] = `
<div
  className="add_frame"
  onClick={[Function]}
  title="shortcut: N"
>
  <div
    className="add_frame-icon"
  />
  <p
    className="add_frame-label"
  >
    Add new frame
  </p>
</div>
`;
